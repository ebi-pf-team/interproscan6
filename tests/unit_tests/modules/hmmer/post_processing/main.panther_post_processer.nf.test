nextflow_process {

    name "Test Process PANTHER_POST_PROCESSER"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "PANTHER_POST_PROCESSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/panther/panther_out"
        symlink "tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/panther/panther_out")
                input[1] = [
                    "path-to-bin",
                    "path-to-site_annotaiton-file"
                ]
                input[2] = file("meta/tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa")
                input[3] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
