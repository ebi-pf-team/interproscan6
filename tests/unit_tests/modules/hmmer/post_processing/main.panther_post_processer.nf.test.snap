{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "treegrafter_processed_panther_hits:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T10:25:28.435079"
    }
}